src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -1056,25 +1066,39 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 }
 
             }
-        
+
+            Version version = null;
+            // Try get dependency version
+            if (obj["Version"] != null)
+            {
+                var versionString = obj["Version"].Value<string>();
+                if (!Version.TryParse(versionString, out version))
+                {
+                    logger.LogWarning(
+                        string.Format("The Version of Dependency: {0}, ReferenceType: {1} could not be deserialized.", name, parsedType),
+                        Logging.WarningLevel.Moderate);
+                }
+            }
+
             INodeLibraryDependencyInfo depInfo;
             //select correct constructor based on referenceType
             switch (parsedType)
             {
                 case ReferenceType.Package:
                     depInfo = new PackageDependencyInfo(name, version);
                     break;
-                /*TODO add other cases: for example
-                 * case ReferenceType.ZeroTouch
-                 * depInfp = new ZeroTouchDependencyInfo(name,version)
-                */
+                case ReferenceType.DYFFILE:
+                    depInfo = new LocalDefinitionInfo(name, ReferenceType.DYFFILE);
+
+                    break;
+                case ReferenceType.ZeroTouch:
+                    depInfo = new LocalDefinitionInfo(name, ReferenceType.ZeroTouch);
+                    break;
                 default:
-                    depInfo = new PackageDependencyInfo(name, version);
+                    depInfo = new LocalDefinitionInfo(name);
```

Comment:
Can you specify the reason for this change?

Reply:
We used to have only PackageDependencyInfo before so it was in the default field. Now we have added cases for DYFFILE and ZeroTouch along with Package reference types. So I was defaulting to LocalDefinitionInfo instead. We can remove this once all cases are addressed.

Reply:
Thanks!